tiff-2-png-converter.py

```diff
@@ -7,100 +7,249 @@
+
+    return image_out
+
+
+def save_rgb_png_image(volume, width, height, img_type):
+    image_out = build_image_sequence(volume, width, height, img_type)
+    name, extension = os.path.splitext(arg_values["dest"])
+    if not extension:
+        extension = ".png"
+
+    amax = np.amax(image_out)
+    norm = np.zeros((image_out.shape))
+    normalized_image_u16bit = cv2.normalize(
+        image_out, norm, 0, 2 ** 16, cv2.NORM_MINMAX
+    )
+    numpngw.write_png(name + extension, image_out)
+    numpngw.write_png(name + "_normalized" + extension, normalized_image_u16bit)
+
+    file = open(name + "_metadata", "a")
+    file.write("Width:" + str(width) + "\n")
+    file.write("Heigth:" + str(height) + "\n")
+    file.write("Depth:" + str(arg_values["nimgs"]) + "\n")
+    file.write("bps:" + str(image_out.dtype) + "\n")
+    file.write("amax:" + str(amax) + "\n")
+    file.close()
+    logging.info("Image written to file-system : " + name + extension)
+
+
+@app.command()
+def convert_to_png(
+    src: str = typer.Argument(..., help="Path to source folder"),
+    dest: str = typer.Argument(..., help="Path where png will be saved"),
+    nimgs: int = typer.Option(
+        None,
+        help=" Number of images in your stack. It will take all the files in the src folder by default",
+    ),
+    verbose: bool = typer.Option(
+        False,
+        help=" Output log messages on console. By default it will print to .log file",
+    ),
+    merge: bool = typer.Option(
+        False,
+        help=" Merge the channels into a single rgb image. False by default, creating an image png per channel",
+    ),
+    gamma: float = typer.Option(
+        None,
+        help=" Applies gamma correction to the dataset. Set a value > 0. Off by default",
+    ),
+    equilize_histogram: str = typer.Option(
+        None, help=" Options: 'opencv' or 'custom' to select implementation"
+    ),
+):
+    arg_values["src"] = src
+    arg_values["dest"] = dest
+    arg_values["nimgs"] = nimgs
+    arg_values["verbose"] = verbose
+    arg_values["merge"] = merge
+    arg_values["gamma"] = gamma
+    arg_values["equilize_histogram"] = equilize_histogram
+
+    name, extension = os.path.splitext(dest)
+    if not (os.path.exists(src) and os.path.exists(os.path.dirname(name))):
+        raise ValueError("ERROR: Verify source and destination paths exists")
+
+    if equilize_histogram is not None:
+        if equilize_histogram == "opencv":
+            arg_values[
+                "equalize_histogram_method"
+            ] = iph.equalize_image_histogram_opencv
+        elif equilize_histogram == "custom":
+            arg_values[
+                "equalize_histogram_method"
+            ] = iph.equalize_image_histogram_custom
+        else:
+            raise ValueError("ERROR: equilize_histogram value is not valid")
+
+    list_files, width, heigth = list_folder_files(src)
+    if nimgs is None:
+        nimgs = len(list_files)
+    image_r = [None] * nimgs
```

Comment:
Again ... I thought we are basically not using RGB. the data type of the input image is the data type of the output image.